socorro/unittest/external/postgresql/test_adi.py

```diff
@@ -0,0 +1,143 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import datetime
+
+from nose.plugins.attrib import attr
+from nose.tools import eq_, assert_raises
+
+from socorro.external.postgresql.adi import ADI
+from socorro.external import MissingArgumentError
+
+from unittestbase import PostgreSQLTestCase
+
+
+@attr(integration='postgres')  # for nosetests
+class IntegrationTestADI(PostgreSQLTestCase):
+
+    def setUp(self):
+        """Set up this test class by populating the reports table with fake
+        data. """
+        super(IntegrationTestADI, self).setUp()
+        self._truncate()
+        cursor = self.connection.cursor()
+
+        self.date = datetime.datetime(2015, 7, 1)
+        yesterday = self.date.replace(hour=0) - datetime.timedelta(hours=24)
+        products = ('Firefox', 'Thunderbird')
+        versions = ('39', '40')
+        platforms = ('Linux', 'Darwin')
+        channels = ('release', 'beta')
+        adi_count = 1
+        for product in products:
+            for version in versions:
+                for platform in platforms:
+                    for channel in channels:
+                        cursor.execute("""
+                            INSERT INTO raw_adi (
+                                adi_count,
+                                date,
+                                product_name,
+                                product_os_platform,
+                                product_os_version,
+                                product_version,
+                                build,
+                                product_guid,
+                                update_channel,
+                                received_at
+                            )
+                            VALUES (
+                                %s, %s, %s, %s, %s, %s, %s, %s, %s, now()
+                            )
+                        """, (
+                            adi_count,
+                            yesterday,
+                            product,
+                            platform,
+                            '1.0',
+                            version,
+                            '20140903141017',
+                            '{abc}',
+                            channel,
+                        ))
+                        adi_count *= 2
+        cursor.execute('select count(*) from raw_adi')
+        count, = cursor.fetchone()
+        assert count == 2 * 2 * 2 * 2, count
```

Comment:
Actually the more I look at it I think it just needs a more human-friendly comment.